test/e2e/antreapolicy_test.go

```diff
@@ -2366,12 +2366,17 @@ func testFQDNPolicy(t *testing.T) {
 		SetPriority(1.0).
 		SetAppliedToGroup([]ACNPAppliedToSpec{{NSSelector: map[string]string{}}})
 	builder.AddFQDNRule("*google.com", v1.ProtocolTCP, nil, nil, nil, "r1", nil, crdv1alpha1.RuleActionReject)
-	builder.AddFQDNRule("wayfair.com", v1.ProtocolTCP, nil, nil, nil, "r2", nil, crdv1alpha1.RuleActionDrop)
+	builder.AddFQDNRule("facebook.com", v1.ProtocolTCP, nil, nil, nil, "r2", nil, crdv1alpha1.RuleActionDrop)
+
+	acnp := builder.Get()
+	k8sUtils.CreateOrUpdateACNP(acnp)
+	failOnError(waitForResourceReady(acnp, timeout), t)
+	time.Sleep(networkPolicyDelay)
```

Comment:
I was wondering why it needs both `waitForResourceReady` and `time.Sleep`. `waitForResourceReady`  sounds a deterministic way to wait for policy to be realized and `time.Sleep` sounds a rough way. But we shouldn't need both. 
I took a look at the implementation of `waitForResourceReady` and found it just checks whether the resource exists. I think the simpler way is to just check the returned error of `CreateOrUpdateACNP`.

Reply:
Good question. I was trying to follow the steps here and didn't think too much: https://github.com/antrea-io/antrea/blob/main/test/e2e/antreapolicy_test.go#L2558-L2589. I tried to find the explanation in previous PRs, but couldn't find it. And from my understanding, `CreateOrUpdateACNP`return nil error doesn't mean the ACNP has been enforced in OVS.